hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -269,19 +269,35 @@ public void close() {
 
   /**
    * Clears the partition Map and reset view states.
+   * <p>
+   * NOTE: This method SHOULD NOT BE OVERRIDDEN which may cause stale file system view
+   * to be served.  Instead, override {@link AbstractTableFileSystemView#runReset} to
+   * add custom logic.
    */
   @Override
   public void reset() {
     try {
       writeLock.lock();
-      clear();
-      // Initialize with new Hoodie timeline.
-      init(metaClient, getTimeline());
+      runReset();
     } finally {
       writeLock.unlock();
     }
   }
 
+  /**
+   * Resets the view states, which can be overridden by subclasses.  This reset logic is guarded
+   * by the write lock.
+   * <p>
+   * NOTE: This method SHOULD BE OVERRIDDEN for any custom logic.  DO NOT OVERRIDE
```

Comment:
then make it abstract? the interface design here should force the implementations to do the right thing.

Reply:
No longer needed.